client/utils/spectreFetch.js

```diff
@@ -0,0 +1,43 @@
+/*
```

Comment:
Could we rename this file to `ajax.js` and then change the methods from `SpectrePost`, `SpectreGet` etc to just `Post`, `Get`